test/v2/controllers/CreateAmendForeignPropertyAnnualSubmissionControllerSpec.scala

```diff
@@ -125,108 +66,90 @@ class CreateAmendForeignPropertyAnnualSubmissionControllerSpec
+        runErrorTest(RuleTaxYearNotSupportedError)
       }
+    }
 
-      "service errors occur" should {
-        def serviceErrors(mtdError: MtdError, expectedStatus: Int): Unit = {
-          s"a $mtdError error is returned from the service" in new Test {
-
-            MockAmendForeignPropertyAnnualSubmissionRequestParser
-              .parseRequest(rawData)
-              .returns(Right(request))
-
-            MockAmendForeignPropertyAnnualSubmissionService
-              .amend(request)
-              .returns(Future.successful(Left(ErrorWrapper(correlationId, mtdError))))
-
-            val result: Future[Result] = controller.handleRequest(nino, businessId, taxYear)(fakeRequestWithBody(requestJson))
-
-            status(result) shouldBe expectedStatus
-            contentAsJson(result) shouldBe Json.toJson(mtdError)
-            header("X-CorrelationId", result) shouldBe Some(correlationId)
-
-            val auditResponse: AuditResponse = AuditResponse(expectedStatus, Some(Seq(AuditError(mtdError.code))), None)
-            MockedAuditService.verifyAuditEvent(event(auditResponse)).once
-          }
-        }
-
-        val input = Seq(
-          (NinoFormatError, BAD_REQUEST),
-          (BusinessIdFormatError, BAD_REQUEST),
-          (TaxYearFormatError, BAD_REQUEST),
-          (RuleTypeOfBusinessIncorrectError, BAD_REQUEST),
-          (RuleTaxYearNotSupportedError, BAD_REQUEST),
-          (RulePropertyIncomeAllowanceError, BAD_REQUEST),
-          (RuleDuplicateCountryCodeError, BAD_REQUEST),
-          (NotFoundError, NOT_FOUND),
-          (InternalError, INTERNAL_SERVER_ERROR),
-          (RuleIncorrectGovTestScenarioError, BAD_REQUEST)
-        )
+    "service errors occur" should {
+      "the service returns an error" in new Test {
+        MockAmendForeignPropertyAnnualSubmissionRequestParser
+          .parseRequest(rawData)
+          .returns(Right(request))
+
+        MockAmendForeignPropertyAnnualSubmissionService
+          .amend(request)
+          .returns(Future.successful(Left(ErrorWrapper(correlationId, InternalError))))
 
-        input.foreach(args => (serviceErrors _).tupled(args))
+        runErrorTest(InternalError)
       }
     }
   }
 
+  trait Test extends ControllerTest with AuditEventChecking[GenericAuditDetail] {
+
+    val controller = new CreateAmendForeignPropertyAnnualSubmissionController(
+      authService = mockEnrolmentsAuthService,
+      lookupService = mockMtdIdLookupService,
+      parser = mockCreateAmendForeignPropertyAnnualSubmissionRequestParser,
+      service = mockService,
+      auditService = mockAuditService,
+      hateoasFactory = mockHateoasFactory,
+      cc = cc,
+      idGenerator = mockIdGenerator
+    )
+
+    protected def callController(): Future[Result] = controller.handleRequest(nino, businessId, taxYear)(fakeRequestWithBody(requestJson))
+
+    protected val requestJson: JsValue = createAmendForeignPropertyAnnualSubmissionRequestBodyMtdJson
```

Comment:
can we move this down so it's with all the other data groups (so to around line 144/145)?